### PR TITLE
Updated readme with alternate install

### PR DESCRIPTION
**Pull request by rdmueller**
As I had problems to get asciidoctor-pdf.js up and running on my machine, I thought it would make sense to add the necessary steps which helped me to succeed.
Thanx to @ahus1 for the proposed solution! (issue #97)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -90,6 +90,39 @@ Runtime Environment (node v10.15.1 on linux)
 CLI version 3.0.1
 ```
 
+## Install locally via yarn (alternative install)
+
+required: current version of yarn installed (should also work with npm)
+
+create a file `package.json`within your project with the following content
```
Comment (ahus1) [inline README.md]: There's a blank missing after the second backtick (`). Consider a capital "C" at the beginning, and maybe a colon at the end?

```suggestion
Create a file `package.json` within your project with the following content:
```

File: README.md [hunk 2]
```diff
@@ -90,6 +90,39 @@ Runtime Environment (node v10.15.1 on linux)
...
+  },
+  "scripts": {
+    "render-pdf": "asciidoctor-pdf <path to your adoc-file>"
+  }
+}
+```
+
+the commands
+
+```javascript
```
Comment (ahus1) [inline README.md]: This is no JavaScript, actually a shell script

`````suggestion
```
`````

File: README.md [hunk 3]
```diff
@@ -90,6 +90,39 @@ Runtime Environment (node v10.15.1 on linux)
...
+    "asciidoctor": "^2.0.3",
+    "asciidoctor-pdf": "^1.0.0-alpha.3"
+  },
+  "scripts": {
+    "render-pdf": "asciidoctor-pdf <path to your adoc-file>"
+  }
+}
+```
+
+the commands
```
Comment (ahus1) [inline README.md]: Starting a new sentence here, therefore capital case?

```suggestion
The commands
```

File: README.md [hunk 4]
```diff
@@ -90,6 +90,39 @@ Runtime Environment (node v10.15.1 on linux)
 CLI version 3.0.1
 ```
 
+## Install locally via yarn (alternative install)
+
+required: current version of yarn installed (should also work with npm)
```
Comment (ahus1) [inline README.md]: Consider capital "R" as this starts a new 

```suggestion
Required: current version of yarn or npm installed
```

File: README.md [hunk 5]
```diff
@@ -90,6 +90,39 @@ Runtime Environment (node v10.15.1 on linux)
 CLI version 3.0.1
 ```
 
+## Install locally via yarn (alternative install)
```
Comment (ahus1) [inline README.md]:
```suggestion
## Install locally via npm or yarn (alternative install)
```

File: README.md (hunk 2 above)
Comment (rdmueller) [reply README.md]: argh. copy and paste...

**Comment by rdmueller** [comment]
@ahus1 thanx for all those fixes!

File: README.md [hunk 6]
```diff
@@ -90,6 +90,39 @@ Runtime Environment (node v10.15.1 on linux)
...
+```
+yarn install
+yarn render-pdf
+```
+
+will then install `asciidoctor` and `asciidoctor-pdf` as local module and render your AsciiDoc file as PDF to the same folder as your soruce file is located.
+
+To use npm instead of yarn, use `npm install` and `npm run render-pdf` instead.
+
+Note: this is verified to work on Windows in WSL.
```
Comment (Mogztter) [inline README.md]: Admonition does not exist in Markdown but could you please use the following syntax for consistency:

```suggestion
**NOTE:** This is verified to work on Windows in WSL.
```

File: README.md [hunk 7]
```diff
@@ -90,6 +90,39 @@ Runtime Environment (node v10.15.1 on linux)
 CLI version 3.0.1
 ```
 
+## Install locally via npm or yarn (alternative install)
+
+Required: current version of yarn or npm installed
```
Comment (Mogztter) [inline README.md]: I think we should only explain how to do it with `npm`.
The reason is that `npm` is bundled with Node.js. So if you are new to the Node.js ecosystem, you will learn about `node` and `npm`. At this point, I strongly believe that introducing another tool can be confusing. What is yarn? Do I have to use it? How do I install it? What is the difference with npm? ...

I also believe that if you are using `yarn`, you are already familiar with the Node.js ecosystem and with `npm`. So you already know how to "translate" a npm command to a yarn command.
We could add a note to explicitly state that the user may also use `yarn` but it pretty much comes without saying.

We can take some inspiration from Antora install guide: https://docs.antora.org/antora/2.2/install/install-antora/

Comment (ahus1) [reply README.md]: +1

**Comment by Mogztter** [comment]
I've updated the documentation.
I don't want to explain the whole create a `package.json`. I think we should create a dedicated package for that, see: https://docs.npmjs.com/cli/init

So basically a third install option will be:

```
$ npm init asciidoctor-pdf-project ./my-asciidoctor-pdf-project
```

File: README.md (hunk 7 above)
Comment (rdmueller) [reply README.md]: To be honest, the yarn example helped me a lot to get things up and running.
I do not work with node, npm or yarn on a daily basis and you've seen how I struggled today with the installation. I even had this problem with installing the core module because I just did a try and error approach.
Today I also removed and installed npm/node several times. My experience with node and npm on windows is that you can easily get it wrong. My experience with yarn is better and it is just great (in a situation like mine) to have an alternate path to get things done.

tl;dr: yes, it makes sense to turn things around and describe the local install with npm and mention yarn as second option. But I wouldn't remove the yarn example. But it is up to you to decide.

**Comment by rdmueller** [comment]
the current version looks good to me. I guess the yarn reference would have solved my problem.

File: README.md (hunk 7 above)
Comment (Mogztter) [reply README.md]: > To be honest, the yarn example helped me a lot to get things up and running.

I don't see how installing an extra package made things easier for you? You still have to install node (+npm) to install yarn right? Could you please elaborate?

I think what would help someone with no prior experience with node, npm or yarn is the `npm init` command. With that you will be able to initialize everything and the packages will be installed locally (so hopefully no permissions errors).

The complete install procedure would be:

```
$ npm init asciidoctor-pdf-project ./my-asciidoctor-pdf-project
$ cd my-asciidoctor-pdf-project
$ npm run convert
```

The last command will convert a sample file to PDF.

File: README.md (hunk 6 above)
Comment (rdmueller) [reply README.md]: btw: why don't you convert the `README.md` to `README.adoc`?

File: README.md (hunk 7 above)
Comment (rdmueller) [reply README.md]: now you reveal my whole JavaScript incompetence with your clever questions :-)
Since I didn't know that yarn is another JavaScript package, I insalled it via `sudo apt-get install yarn` :-)

But you are right - your tool is not only interesting to people with a ajvascript background, but also to people with absolutly no JavaScript backroung (even when it is hard to imagine that such people still exist these days ;-)

Comment (Mogztter) [reply README.md]: > now you reveal my whole JavaScript incompetence with your clever questions :-)
Since I didn't know that yarn is another JavaScript package, I insalled it via sudo apt-get install yarn :-)

It makes sense. I didn't thought about system package manager.
I still think the root cause is the global installation without nvm (ie. installation in the system scope with sudo).
As far as I know Puppeteer does not install well with sudo/global flag: 
- https://github.com/GoogleChrome/puppeteer/issues/4574
- https://github.com/GoogleChrome/puppeteer/issues/375

We should probably add a note on `nvm`: https://docs.antora.org/antora/2.2/install/linux-requirements/#install-nvm 

@ahus1  Out of curiosity, are you using `nvm` ?

> But you are right - your tool is not only interesting to people with a ajvascript background, but also to people with absolutly no JavaScript backroung (even when it is hard to imagine that such people still exist these days ;-)

Yes and that's why I want to provide a reliable and straight forward installation procedure but that's not an easy task :grimacing:

File: README.md (hunk 6 above)
Comment (Mogztter) [reply README.md]: I wish... but npmjs.org only supports Markdown :disappointed:

File: README.md (hunk 7 above)
Comment (Mogztter) [reply README.md]: @ahus1 @rdmueller Did you try to install on Windows without using WSL?

**Comment by Mogztter** [comment]
I've added a "Requirements" section heavily inspired by Antora: https://github.com/Mogztter/asciidoctor-pdf.js/blob/fb16c724f4b95a90b77f81dad76c1988ac525366/README.md#requirements

File: README.md (hunk 7 above)
Comment (rdmueller) [reply README.md]: I just started to install node.js on windows... let's see...

Comment (rdmueller) [reply README.md]: here is the result. I installed node.js through the .msi package - the method which is described on the node.js homepage:

```
PS ralfd> npm install -g asciidoctor asciidoctor-pdf
C:\Users\ralfd\AppData\Roaming\npm\asciidoctorjs -> C:\Users\ralfd\AppData\Roaming\npm\node_modules\asciidoctor\bin\asciidoctor
C:\Users\ralfd\AppData\Roaming\npm\asciidoctor -> C:\Users\ralfd\AppData\Roaming\npm\node_modules\asciidoctor\bin\asciidoctor
C:\Users\ralfd\AppData\Roaming\npm\asciidoctor-pdf -> C:\Users\ralfd\AppData\Roaming\npm\node_modules\asciidoctor-pdf\bin\asciidoctor-pdf
npm WARN asciidoctor-pdf@1.0.0-alpha.3 requires a peer of @asciidoctor/core@^2.0.0-rc.1 but none is installed. You must install peer dependencies yourself.
npm WARN @asciidoctor/cli@3.0.1 requires a peer of @asciidoctor/core@^2.0.0-rc.1 but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules\asciidoctor-pdf\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

+ asciidoctor@2.0.3
+ asciidoctor-pdf@1.0.0-alpha.3
updated 2 packages in 25.078s
PS ralfd> asciidoctor-pdf -v
internal/modules/cjs/loader.js:797
    throw err;
    ^

Error: Cannot find module '@asciidoctor/core'
Require stack:
- C:\Users\ralfd\AppData\Roaming\npm\node_modules\asciidoctor-pdf\node_modules\@asciidoctor\cli\lib\options.js
- C:\Users\ralfd\AppData\Roaming\npm\node_modules\asciidoctor-pdf\node_modules\@asciidoctor\cli\lib\cli.js
- C:\Users\ralfd\AppData\Roaming\npm\node_modules\asciidoctor-pdf\lib\cli.js
- C:\Users\ralfd\AppData\Roaming\npm\node_modules\asciidoctor-pdf\bin\asciidoctor-pdf
[90m    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:794:15)[39m
[90m    at Function.Module._load (internal/modules/cjs/loader.js:687:27)[39m
[90m    at Module.require (internal/modules/cjs/loader.js:849:19)[39m
[90m    at require (internal/modules/cjs/helpers.js:74:18)[39m
    at Object.<anonymous> (C:\Users\ralfd\AppData\Roaming\npm\node_modules\[4masciidoctor-pdf[24m\node_modules\[4m@asciidoctor[24m\cli\lib\options.js:3:21)
[90m    at Module._compile (internal/modules/cjs/loader.js:956:30)[39m
[90m    at Object.Module._extensions..js (internal/modules/cjs/loader.js:973:10)[39m
[90m    at Module.load (internal/modules/cjs/loader.js:812:32)[39m
[90m    at Function.Module._load (internal/modules/cjs/loader.js:724:14)[39m
[90m    at Module.require (internal/modules/cjs/loader.js:849:19)[39m {
  code: [32m'MODULE_NOT_FOUND'[39m,
  requireStack: [
    [32m'C:\\Users\\ralfd\\AppData\\Roaming\\npm\\node_modules\\asciidoctor-pdf\\node_modules\\@asciidoctor\\cli\\lib\\options.js'[39m,
    [32m'C:\\Users\\ralfd\\AppData\\Roaming\\npm\\node_modules\\asciidoctor-pdf\\node_modules\\@asciidoctor\\cli\\lib\\cli.js'[39m,
    [32m'C:\\Users\\ralfd\\AppData\\Roaming\\npm\\node_modules\\asciidoctor-pdf\\lib\\cli.js'[39m,
    [32m'C:\\Users\\ralfd\\AppData\\Roaming\\npm\\node_modules\\asciidoctor-pdf\\bin\\asciidoctor-pdf'[39m
  ]
}
```

Comment (rdmueller) [reply README.md]: now I will try to install it through nvm...

Comment (rdmueller) [reply README.md]: same problem with an nvm install...

Comment (Mogztter) [reply README.md]: Could you please list the packages/directories in the `C:\Users\ralfd\AppData\Roaming\npm\node_modules` directory?

The package `@asciidoctor/core` is a direct dependency of `asciidoctor` and should be installed... I will try the install procedure on a Windows machine.

And also could you please execute the following commands: `npm --version` and `node --version`?

> same problem with an nvm install...

You've successfully installed `nvm`, `node` and `npm`? But the command `npm install ...` throws the same error, correct?

Comment (Mogztter) [reply README.md]: I don't reproduce this issue on Linux but I still get warnings when installing:

```
npm WARN asciidoctor-pdf@1.0.0-alpha.3 requires a peer of @asciidoctor/core@^2.0.0-rc.1 but none is installed. You must install peer dependencies yourself.
npm WARN @asciidoctor/cli@3.0.1 requires a peer of @asciidoctor/core@^2.0.0-rc.1 but none is installed. You must install peer dependencies yourself.
```

Could you try:

```
npm i -g @asciidoctor/core asciidoctor-pdf
```

Comment (Mogztter) [reply README.md]: I can reproduce this issue. For some reason `npm i -g asciidoctor asciidoctor-pdf` does not work on Windows but `npm i -g @asciidoctor/core asciidoctor-pdf` does.

File: README.md [hunk 8]
```diff
@@ -81,6 +114,8 @@ Verify that the `asciidoctor-pdf` command is available on your `PATH` by running
 
     $ asciidoctor-pdf --version
 
+**NOTE:** If you get an error about [Executions Policies](https://go.microsoft.com/fwlink/?LinkID=135170) when running this command on PowerShell, try to use the following command instead: `$ asciidoctor-pdf.cmd --version`.
```
Comment (Mogztter) [inline README.md]: @rdmueller When using `nvm` to install `npm` and `node`, I got an error about executions policies. The command `asciidoctor-pdf --version` will in fact execute `asciidoctor-pdf.ps1 --version` and for whatever reason executing `ps1` file is not allowed in PowerShell on my machine.

Anyway, with the updated install command it should work: `npm i -g @asciidoctor/core asciidoctor-pdf`

**Comment by Mogztter** [comment]
I think it's good to go, we have added a lot of information about Node, nvm and npm. I've successfully installed `asciidoctor-pdf`` on Windows 10 following the instructions.

File: README.md (hunk 8 above)
Comment (rdmueller) [reply README.md]: I just also succeeded with 
```
    npm i -g @asciidoctor/core asciidoctor-pdf
```

the execution of `ps1` files wasn't a problem for me, because I already removed this restriction a long time ago :-)

**Comment by rdmueller** [comment]
wow. this turned out to be quite a big change, but I guess it will help many users.
Thanx for your time and help - I learned a lot!

**Comment by Mogztter** [comment]
> wow. this turned out to be quite a big change, but I guess it will help many users.

Yes, thanks to you!

> Thanx for your time and help - I learned a lot!

:+1:
